src/components/user-form/UserForm.jsx

```diff
@@ -1,28 +1,25 @@
 import React, { useCallback, useState } from 'react';
 import PropTypes from 'prop-types';
 import Input from '../input/Input';
+import Form from '../form/Form';
 import Button, { ButtonType } from '../button/Button';
 import styles from './UserForm.module.css';
 
-export default function UserForm({
-  initialValues,
-  onSubmit,
-}) {
-  const [username, setUsername] = useState(initialValues ? initialValues.username : '');
+export default function UserForm({ initialValues, onSubmit }) {
+  const [username, setUsername] = useState(
+    initialValues ? initialValues.username : '',
+  );
 
   const onChange = useCallback(({ target }) => {
     setUsername(target.value);
   }, []);
 
-  const internalOnSubmit = useCallback((e) => {
-    e.preventDefault();
-    onSubmit({ username });
-  }, [onSubmit, username]);
-
   return (
-    <form
+    <Form
       className={styles.form}
-      onSubmit={internalOnSubmit}
+      onSubmit={() => {
```

Comment:
Was the `useCallback` declaration useless in your opinion?

Reply:
Yep, `useCallback` returns a memoized version of the callback that you define within its body;
doing so you avoid React to use a different function (since they are just anonymous) at every rerender.
I propose to leave it!

Reply:
Done!